overlord/servicestate/quota_control.go

```diff
@@ -473,5 +476,46 @@ func UpdateQuota(st *state.State, name string, updateOpts QuotaGroupUpdate) erro
 	}
 
 	// ensure service states are updated
-	return ensureSnapServicesForGroup(st, grp, allGrps)
+	return ensureSnapServicesForGroup(st, grp, allGrps, nil)
+}
+
+// RemoveSnapFromQuota removes the specified snap from the quota.
+// XXX: this should go away and just become an option to UpdateQuota but that is
+// more complexity than we can handle right now
+func RemoveSnapFromQuota(st *state.State, group, snap string) error {
+	allGrps, err := AllQuotas(st)
+	if err != nil {
+		return err
+	}
+
+	// ensure that the quota group exists
+	grp, ok := allGrps[group]
+	if !ok {
+		return fmt.Errorf("quota group %q does not exist", group)
```

Comment:
if we keep RemoveSnapQuota around as well as is,  this error path should be tested

Reply:
I removed RemoveSnapFromQuota and instead reimplemented as EnsureSnapAbsentFromQuotaGroup. It would be slightly nicer as something like EnsureSnapAbsentFromAnyQuotaGroup, but that kinda implies you could have a snap in multiple quota groups which you can't. Happy to hear suggestions on the name, I thought Maybe was kinda weird IMO